File: api/src/main/java/org/openmrs/ProgramWorkflowState.java
/**
 * This Source Code Form is subject to the terms of the Mozilla Public License,
 * v. 2.0. If a copy of the MPL was not distributed with this file, You can
 * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
 * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
 *
 * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
 * graphic logo is a trademark of OpenMRS Inc.
 */
package org.openmrs;


import jakarta.persistence.AttributeOverride;
import jakarta.persistence.AttributeOverrides;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import org.hibernate.envers.Audited;

/**
 * ProgramWorkflowState
 */
@Entity
@Table(name = "program_workflow_state")
@AttributeOverrides({
	@AttributeOverride(name = "name", column = @Column(name = "name", nullable = true, length = 255))
})
@Audited
public class ProgramWorkflowState extends BaseChangeableOpenmrsMetadata {
	
	private static final long serialVersionUID = 1L;
	
	// ******************
	// Properties
	// ******************
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "program_workflow_state_id")
	private Integer programWorkflowStateId;

	@ManyToOne(optional = false)
	@JoinColumn(name = "program_workflow_id", nullable = false)
	private ProgramWorkflow programWorkflow;

	@ManyToOne(optional = false)
	@JoinColumn(name = "concept_id", nullable = false)
	private Concept concept;
	
	@Column(name = "initial", nullable = false, length = 1)
	private Boolean initial;
	
	@Column(name = "terminal", nullable = false, length = 1)
	private Boolean terminal;
	
	// ******************
	// Constructors
	// ******************
	
	/** Default Constructor */
	public ProgramWorkflowState() {
	}
	
	/** Constructor with id */
	public ProgramWorkflowState(Integer programWorkflowStateId) {
		setProgramWorkflowStateId(programWorkflowStateId);
	}
	
	// ******************
	// Instance methods
	// ******************
	
	/** @see Object#toString() */
	@Override
	public String toString() {
		return "State " + getConcept().toString() + " initial=" + getInitial() + " terminal=" + getTerminal();
	}
	
	// ******************
	// Property Access
	// ******************
	
	public Concept getConcept() {
		return concept;
	}
	
	public void setConcept(Concept concept) {
		this.concept = concept;
	}
	
	public Boolean getInitial() {
		return initial;
	}
	
	public void setInitial(Boolean initial) {
		this.initial = initial;
	}
	
	public Boolean getTerminal() {
		return terminal;
	}
	
	public void setTerminal(Boolean terminal) {
		this.terminal = terminal;
	}
	
	public ProgramWorkflow getProgramWorkflow() {
		return programWorkflow;
	}
	
	public void setProgramWorkflow(ProgramWorkflow programWorkflow) {
		this.programWorkflow = programWorkflow;
	}
	
	public Integer getProgramWorkflowStateId() {
		return programWorkflowStateId;
	}
	
	public void setProgramWorkflowStateId(Integer programWorkflowStateId) {
		this.programWorkflowStateId = programWorkflowStateId;
	}
	
	/**
	 * @since 1.5
	 * @see org.openmrs.OpenmrsObject#getId()
	 */
	@Override
	public Integer getId() {
		return getProgramWorkflowStateId();
	}
	
	/**
	 * @since 1.5
	 * @see org.openmrs.OpenmrsObject#setId(java.lang.Integer)
	 */
	@Override
	public void setId(Integer id) {
		setProgramWorkflowStateId(id);
		
	}
}

File: api/src/main/kotlin/org/openmrs/ProgramWorkflowState.kt
/**
 * This Source Code Form is subject to the terms of the Mozilla Public License,
 * v. 2.0. If a copy of the MPL was not distributed with this file, You can
 * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
 * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
 *
 * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
 * graphic logo is a trademark of OpenMRS Inc.
 */
package org.openmrs

import org.hibernate.envers.Audited

/**
 * ProgramWorkflowState
 */
@Audited
open class ProgramWorkflowState : BaseChangeableOpenmrsMetadata {
    
    companion object {
        private const val serialVersionUID = 1L
    }
    
    // ******************
    // Properties
    // ******************
    
    open var programWorkflowStateId: Int? = null
    
    open var programWorkflow: ProgramWorkflow? = null
    
    open var concept: Concept? = null
    
    open var initial: Boolean? = null
    
    open var terminal: Boolean? = null
    
    // ******************
    // Constructors
    // ******************
    
    /** Default Constructor */
    constructor()
    
    /** Constructor with id */
    constructor(programWorkflowStateId: Int?) {
        this.programWorkflowStateId = programWorkflowStateId
    }
    
    // ******************
    // Instance methods
    // ******************
    
    /** @see Object.toString */
    override fun toString(): String {
        return "State ${concept.toString()} initial=$initial terminal=$terminal"
    }
    
    // ******************
    // Property Access
    // ******************
    
    /**
     * @since 1.5
     * @see org.openmrs.OpenmrsObject.getId
     */
    override var id: Int?
        get() = programWorkflowStateId
        set(id) {
            programWorkflowStateId = id
        }
}
